File: c#/money-problem.Domain/MissingExchangeRatesException.cs
namespace money_problem.Domain
{
    public class MissingExchangeRatesException : Exception
    {
        public MissingExchangeRatesException(string[] missingCurrencies)
            : base($"Missing exchange rate(s): [{string.Join(",", missingCurrencies)}]")
        {
        }
    }
}

File: java/src/main/java/money_problem/domain/MissingExchangeRatesException.java
package money_problem.domain;

import io.vavr.collection.List;

public class MissingExchangeRatesException extends Exception {
    public MissingExchangeRatesException(List<String> missingExchangeRates) {
        super(missingExchangeRates.mkString("Missing exchange rate(s): [", ",", "]"));
    }
}
